src/main/java/com/nha/abdm/wrapper/hiu/hrp/consent/HIUConsentService.java

```diff
@@ -0,0 +1,176 @@
+/* (C) 2024 */
+package com.nha.abdm.wrapper.hiu.hrp.consent;
+
+import com.nha.abdm.wrapper.common.RequestManager;
+import com.nha.abdm.wrapper.common.responses.FacadeResponse;
+import com.nha.abdm.wrapper.hip.hrp.link.hipInitiated.responses.GatewayGenericResponse;
+import com.nha.abdm.wrapper.hiu.hrp.consent.requests.ConsentStatusRequest;
+import com.nha.abdm.wrapper.hiu.hrp.consent.requests.FetchConsentRequest;
+import com.nha.abdm.wrapper.hiu.hrp.consent.requests.InitConsentRequest;
+import com.nha.abdm.wrapper.hiu.hrp.consent.requests.OnNotifyRequest;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import reactor.core.Exceptions;
+
+public class HIUConsentService implements HIUConsentInterface {
+
+  private static final Logger log = LogManager.getLogger(HIUConsentService.class);
+
+  @Value("${consentInitPath}")
+  private String consentInitPath;
+
+  @Value("${consentStatusPath}")
+  private String consentStatusPath;
+
+  @Value("${consentOnNotifyPath}")
+  private String consentOnNotifyPath;
+
+  @Value("${fetchConsentPath}")
+  private String fetchConsentPath;
+
+  private final RequestManager requestManager;
+
+  @Autowired
+  public HIUConsentService(RequestManager requestManager) {
+    this.requestManager = requestManager;
+  }
+
+  @Override
+  public ResponseEntity<FacadeResponse> initiateConsentRequest(
+      InitConsentRequest initConsentRequest) {
+    try {
+      ResponseEntity<GatewayGenericResponse> response =
+          requestManager.fetchResponseFromGateway(consentInitPath, initConsentRequest);
+      if (Objects.nonNull(response.getBody())
+          && Objects.nonNull(response.getBody().getErrorResponse())) {
```

Comment:
The other way of checking the error could be based on the `response.getStatusCode() != 2xx` then return an error (I am hoping the status codes are properly set by Gateway). Also, I see this code is repeated in this class, may be we can refactor this to reduce the number of lines in the class.

Reply:
Have created an issue to handle this: https://github.com/NHA-ABDM/ABDM-wrapper/issues/46